lib/flappi/builder_factory.rb

```diff
@@ -281,7 +281,8 @@ def self.render_response_json(controller)
       if response_object.respond_to?(:status_code)
         error_info = response_object.status_error_info
         response_hash = error_info.is_a?(String) ? { error: error_info } : { errors: error_info }
-        controller.render json: response_hash.to_json, plain: error_info, status: response_object.status_code
+        Flappi::Utils::Logger.i "flappi error response: #{response_hash.inspect}"
```

Comment:
Maybe log `response_object` instead of just `response_hash`..?

---

Looking at it myself, seems weird.  I'd think we could trace it without Flappi, eg. `response_object.status_error_info` is being set from `@status_error_info`, which is set when we call `return_error(422, 'Error message')` in Investapp in the API Controller stack.

I can't see a `return_error(422, nil)` or anything like I expect…

---

Maybe: this is sometimes called with a named argument…does this work?

[investapp/app/services/api/trades/crud_service.rb#256](https://github.com/sharesight/investapp/blob/master/app/services/api/trades/crud_service.rb#L256)
[investapp/app/services/api/trades/crud_service.rb#330](https://github.com/sharesight/investapp/blob/master/app/services/api/trades/crud_service.rb#L330)
_Also, this seems like a very weird error to throw in a trade endpoint._
```rails
return_error(422, portfolio_id: 'Portfolio not valid')
```

---

Defined in Flappi as:
[flappi/lib/flappi/response_builder.rb#287…290](https://github.com/sharesight/flappi/blob/be519343dc8e60be0162fd496c24ae7c8e197a04/lib/flappi/response_builder.rb#L287…L290)

Maybe this should be slightly different, like so…?  I'm unsure.
```diff
-def return_error(status_code, error_info)
+def return_error(status_code, *error_info)
```

Reply:
@kylorhall Thanks for looking a bit deeper. Some comments:

- The check for a valid portfolio is actually just a check for deletion. And it's fine not to accept trades for a deleted portfolio. But the error message can be improved.
- Expecting only a code and one error object for `return_error` is fine. Normally it's a hash or a string. I don't see a use case for more than one parameter so far.
- I didn't take the `response_object` for two reasons: 1. I don't know what this object is and what a `.inspect` would like (maybe it's just the object id which is not very helpfull), depending on the implementation of the `inspect` method, and 2. I want to see what the API client sees. If that isn't very helpful we improve it.